kof-operator/internal/controller/clusterdeployment_kof_cluster_role.go

```diff
@@ -115,18 +120,21 @@ func (r *ClusterDeploymentReconciler) reconcileChildClusterRole(
 		return err
 	}
 
+	if err := r.createProfile(childClusterDeployment, regionalClusterDeployment, ctx); err != nil {
+		log.Error(err, "Failed to create profile")
+		return err
+	}
+
 	configMap := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      getConfigMapName(childClusterDeployment.Name),
 			Namespace: childClusterDeployment.Namespace,
 			OwnerReferences: []metav1.OwnerReference{
 				// Auto-delete ConfigMap when child ClusterDeployment is deleted.
-				{
-					APIVersion: "k0rdent.mirantis.com/v1alpha1",
-					Kind:       "ClusterDeployment",
-					Name:       childClusterDeployment.Name,
-					UID:        childClusterDeployment.GetUID(),
-				},
+				GetOwnerReference(
+					childClusterDeployment.Name,
+					childClusterDeployment.GetUID(),
+				),
```

Comment:
Nice idea, but let's make it even more DRY:
```suggestion
				GetOwnerReference(childClusterDeployment),
```
I'll show implementation in the next suggestion.